app/page.js

```diff
@@ -1,95 +1,106 @@
 
-        <a
-          href="https://vercel.com/new?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
-          className={styles.card}
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          <h2>
-            Deploy <span>-&gt;</span>
-          </h2>
-          <p>
-            Instantly deploy your Next.js site to a shareable URL with Vercel.
-          </p>
-        </a>
-      </div>
+          {result.data && result.data.list && result.data.list.length > 0 && (
+            <>
+              <h3>Temperature</h3>
+              <div style={{ width: '300px', height: '125px', padding: '0', margin: '0' }}>
+                <Sparklines data={result.data.list.slice(0, 5).map(datum => datum.main.temp)} height={50}>
+                  <SparklinesLine color="#FFA500" />
+                </Sparklines>
+              </div>
+              <div className='search-results-five'>
+                <ul>
+                  {result.data.list.slice(0, 5).map((datum, index) => (
+                    <li key={index} className='data'>{datum.main && datum.main.temp}</li>
+                  ))}
+                </ul>
+                <ul>
+                  <li className='data'>5-day average: {calculateAverage(result.data.list, 'temp')}</li>
+                </ul>
+              </div>
+
+              <h3>Pressure</h3>
+              <div style={{ width: '300px', height: '125px', padding: '0', margin: '0' }}>
+                <Sparklines data={result.data.list.slice(0, 5).map(item => item.main.pressure)} height={50}>
+                  <SparklinesLine color="#000080" />
+                </Sparklines>
+              </div>
+              <div className='search-results-five'>
+                <ul>
+                  {result.data.list.slice(0, 5).map((item, index) => (
+                    <li key={index} className='data'>{item.main && item.main.pressure}</li>
+                  ))}
+                </ul>
+                <ul>
+                  <li className='data'>5-day average: {calculateAverage(result.data.list, 'pressure')}</li>
+                </ul>
+              </div>
+
+              <h3>Humidity</h3>
+              <div style={{ width: '300px', height: '125px', padding: '0', margin: '0' }}>
+                <Sparklines data={result.data.list.slice(0, 5).map(item => item.main.humidity)} height={50}>
+                  <SparklinesLine color="#008000" />
+                </Sparklines>
+              </div>
+              <div className='search-results-five'>
+                <ul>
+                  {result.data.list.slice(0, 5).map((item, index) => (
+                    <li key={index} className='data'>{item.main && item.main.humidity}</li>
+                  ))}
+                </ul>
+                <ul>
+                  <li className='data'>5-day average: {calculateAverage(result.data.list, 'humidity')}</li>
+                </ul>
```

Comment:
this html is a bit long and hard to understand, there's logic also that could have been extracted to functions to help with readability